components/gitpod-protocol/src/messaging/error.ts

```diff
@@ -24,7 +23,7 @@ export class ApplicationError extends Error {
 
 export namespace ApplicationError {
     export function hasErrorCode(e: any): e is Error & { code: ErrorCode; data?: any } {
-        return e && e.code !== undefined;
+        return ErrorCode.is(e["code"]);
```

Comment:
not all errors with code are ApplicationError, we will consider in the past some typeorm and Node.js errors as AppilcationErorrs, while they should be Internal